keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -786,7 +786,7 @@ class ClusterHandler(
     )
       .also { them ->
         val allSame: Boolean = them.distinctBy { it.launchConfiguration.appVersion }.size == 1
-        val healthy: Boolean = them.all { it.instanceCounts?.isHealthy() == true }
+        val healthy: Boolean = resource.spec.deployWith.considerOnlyAmazonHealth || them.all { it.instanceCounts?.isHealthy() == true }
```

Comment:
`instanceCounts` is entirely based on Discovery. If you look at a server group that doesn't use Discovery, all the counts are zero even if there are instances.

Reply:
Is there some point between the start of the deployment and the end where things change? Like, number of instances? I'm worried that this will consider things healthy when the asg is created, and then we will have a situation like: test deploys, but it's marked as "successfully deployed" the minute the asg is created. Then, we move on to deploying staging, but the orca task isn't finished and the test deployment wasn't actually good.

Reply:
I'm looking into how Orca handles this